store/index-refactor.js

```diff
@@ -100,8 +114,48 @@ export const mutations = {
             Object.fromEntries(p_columns.map((column) => [column, null]));
     },
 
+    initializeDataDictionary(p_state) {
```

Comment:
I agree the second option is clearer - we could see if this is something the linter can handle for us! 

Re the `for` vs `map` conversation: I can live with both. To me a short `map` statement is clearer than a for loop because it has less boilerplate stuff - especially if the goal is to just create another array. This case here would probably make sense to put in a `for` loop. In that case, I would say we stick with [the `for ... of` syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) since that has been around for a while now and is clearer.

Would probably be good to write down some general guidelines on when we go for one over the other since that has come up a couple of times. I'd be OK with using `for ... of` unless its a small mapping or the output is another array.

Reply:
One thing I did notice while looking at this again: I'm pretty sure my mutation [wouldn't be reactive](https://v2.vuejs.org/v2/guide/reactivity.html) the way I wrote it so far. So that'll have to change too.

Reply:
> I would say we stick with [the for ... of syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) since that has been around for a while now and is clearer

Sounds good and good point.

> I'm pretty sure my mutation [wouldn't be reactive](https://v2.vuejs.org/v2/guide/reactivity.html) the way I wrote it so far. So that'll have to change too.

Since `dataTable.annotated` is assigned to a brand new object, wouldn't this be reactive?

Reply:
No, I don't think it would because the nested attributes (i.e. the column names) aren't defined initially. I addressed it in my current commit. Think that'll do it

Reply:
> Would probably be good to write down some general guidelines on when we go for one over the other since that has come up a couple of times. I'd be OK with using for ... of unless its a small mapping or the output is another array.

So yeah - what's happening here is the creation of a new object (for `dataDictionary.annotated`) based on the keys of an array `dataTable[0]`.

I can point out the performativity/computer science of why the loop is preferable as well.

Reply:
> No, I don't think it would because the nested attributes (i.e. the column names) aren't defined initially. I addressed it in my current commit. Think that'll do it

Interesting. I'll read more into that. Looks good!

Reply:
@surchs Ahh okay. I see it.

>Sometimes you may want to assign a number of properties to an existing object, for example using Object.assign() or _.extend(). However, new properties added to the object will not trigger changes. In such cases, create a fresh object with properties from both the original object and the mixin object:

Reply:
@surchs Ahh okay. I see it.

>Sometimes you may want to assign a number of properties to an existing object, for example using Object.assign() or _.extend(). However, new properties added to the object will not trigger changes. In such cases, create a fresh object with properties from both the original object and the mixin object:

```


// instead of Object.assign(this.someObject, { a: 1, b: 2 })
this.someObject = Object.assign({}, this.someObject, { a: 1, b: 2 })

```